### PR TITLE
[inner-2112] implement interrupt thread through interrupt()/shutdownNow()

### PR DESCRIPTION
**Pull request by wenyh1**
Reason:  
  BUG #. or Improve #.  
Type:  
  BUG/Improve  
Influences：  
   fix xx

### CODE REVIEW

File: src/main/java/com/actiontech/dble/util/NameableScheduledThreadPoolExecutor.java [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+        return name;
+    }
+
+    @Override
+    protected void beforeExecute(Thread t, Runnable r) {
+        super.beforeExecute(t, r);
+        if (threadChecker != null) {
+            threadChecker.startExec(t);
+        }
+        if (null != runnableMap) {
```
Comment (LUAgam) [inline src/main/java/com/actiontech/dble/util/NameableScheduledThreadPoolExecutor.java:34]: If the runnableMap here is meaningless, you can delete it.

File: src/main/java/com/actiontech/dble/services/manager/handler/DbGroupHAHandler.java [hunk 2]
```diff
@@ -20,7 +20,7 @@ public final class DbGroupHAHandler {
 
     private static final String DB_NAME_FORMAT = "a-zA-Z_0-9\\-\\."; //nmtoken
     private static final Pattern PATTERN_DH_DISABLE = Pattern.compile("@@disable\\s*name\\s*=\\s*'([" + DB_NAME_FORMAT + "]+)'" +
-            "\\s*(instance\\s*=\\s*'([" + DB_NAME_FORMAT + "\\,]+)')?$", Pattern.CASE_INSENSITIVE);
+            "\\s*(name\\s*=\\s*'([a-zA-Z_0-9\\-]+)')?$", Pattern.CASE_INSENSITIVE);
```
Comment (LUAgam) [inline src/main/java/com/actiontech/dble/services/manager/handler/DbGroupHAHandler.java:23]: What is the meaning of changing to name here?

File: src/main/java/com/actiontech/dble/singleton/ThreadManager.java [hunk 3]
```diff
@@ -0,0 +1,277 @@
...
+                    sbf.append("============== select thread[{" + threadName + "}] ============== start");
+                    sbf.append("\n \"");
+                    sbf.append(thread.getName());
+                    sbf.append("\" ");
+                    sbf.append("#");
+                    sbf.append(thread.getId());
+                    sbf.append(", state: ");
+                    sbf.append(thread.getState());
+                    sbf.append(", stackTrace: ");
+                    StackTraceElement[] st = thread.getStackTrace();
```
Comment (LUAgam) [inline src/main/java/com/actiontech/dble/singleton/ThreadManager.java:246]: need to pay attention to whether it causes STW?

File: src/main/java/com/actiontech/dble/singleton/ThreadManager.java [hunk 4]
```diff
@@ -0,0 +1,277 @@
...
+
+    public static void recoverThreadPool(String threadName) throws Exception {
+        switch (threadName) {
+            case TIMER_WORKER_NAME:
+                if (!DbleServer.getInstance().getTimerExecutor().isShutdown()) {
+                    throw new Exception("threadPool[" + TIMER_WORKER_NAME + "] is not shutdown, no need to recover");
+                }
+                LOGGER.info("manual recover threadPool[{}] ... start ...", TIMER_WORKER_NAME);
+                DbleServer.getInstance().setTimerExecutor(
+                        ExecutorUtil.createFixed(TIMER_WORKER_NAME, 2, ThreadChecker.getInstance()));
```
Comment (LUAgam) [inline src/main/java/com/actiontech/dble/singleton/ThreadManager.java:173]: inconsistent with the initialization method in DbleServer

File: src/main/java/com/actiontech/dble/singleton/Scheduler.java [hunk 5]
```diff
@@ -102,6 +95,10 @@ public void run() {
                     });
                 } catch (RejectedExecutionException e) {
                     ThreadChecker.getInstance().timerExecuteError(e, "dbInstanceOldConsClear()");
+                } catch (Throwable e) {
+                    if (LOGGER.isDebugEnabled()) {
+                        LOGGER.debug("scheduled task dbInstanceOldConsClear() happen exception: ", e);
```
Comment (LUAgam) [inline src/main/java/com/actiontech/dble/singleton/Scheduler.java:100]: debug?

File: src/main/java/com/actiontech/dble/services/manager/handler/DbGroupHAHandler.java (hunk 2 above)
Comment (wenyh1) [reply src/main/java/com/actiontech/dble/services/manager/handler/DbGroupHAHandler.java:23]: get

File: src/main/java/com/actiontech/dble/singleton/ThreadManager.java (hunk 3 above)
Comment (wenyh1) [reply src/main/java/com/actiontech/dble/singleton/ThreadManager.java]: temporarily remove print requirement，subsequent creation of a separate issue

File: src/main/java/com/actiontech/dble/singleton/Scheduler.java (hunk 5 above)
Comment (wenyh1) [reply src/main/java/com/actiontech/dble/singleton/Scheduler.java]: get

**Review by LUAgam — APPROVED** [review]
lgtm

**Pull request merged**